### DISCUSSION TITLE
Can you add a section to the BTCPay Lightning FAQ "How do I enable Lightning Network Wumbo Channels?"

### DISCUSSION THREAD

**Discussion by clarketowson** [btcpayserver/btcpayserver#5774] [General]
Hi Everyone,

I think the BTCPay Lightning FAQ needs a new topics/question "How do I enable Lightning Network Wumbo Channels?"

With the official instructions from the BTCPay Server devs on the best/most optimal method to do this.

Best Regards,

Clarke

**Comment by dennisreimann** [comment]
Wumbo channels are the default in the current versions of implementations we support (LND and Core Lightning).

**Comment by clarketowson** [comment]
Yes that's what I thought too but there's an issue - for people who upgrade their BTCPay Server from an older version wumbo channels are not enabled after the upgrade.  Or at least that's what I have found on my BTCPay Servers.  Node operators are telling me they can't open wumbo channels to my nodes.  I recently upgraded both nodes to the latest versions of BTCPay but wumbo is not enabled on either of them.

If an FAQ was posted specifically about enabling wumbo it would help people who are experiencing this same issue.

Best Regards,

Clarke

**Comment by clarketowson** [comment]
@dennisreimann i'm happy to work with you on this I have a test server setup specifically for this purpose to develop my procedures and documentation for my business.  Please let me know if there is anything I can do to help.

**Comment by dennisreimann** [comment]
### LND 

For [LND](https://docs.lightning.engineering/lightning-network-tools/lnd/lnd.conf) there are two settings, which affect this:

- `protocol.wumbo-channels=false`
- `maxchansize=10000000`

They can be added by [editing the lnd.conf](https://docs.btcpayserver.org/FAQ/LightningNetwork/#how-to-edit-lndconf) like this:


```yaml
version: '3'
services:
  lnd_bitcoin:
    environment:
      LND_EXTRA_ARGS: |
        protocol.wumbo-channels=true
        maxchansize=10000000
```

## Core Lightning

For CLN there is the [`large-channels` setting](https://docs.corelightning.org/reference/lightningd-config#lightning-channel-and-htlc-options), but it is deprecated and ignored as of v23.11

The Core Lightning config can be edited as lnd.conf (see above).

With current versions of the implementations those changes should not be necessary for channels that will be opened. Existing channels won't be affected by these changes, hence I don't think that editing the configs makes much sense.

**Comment by clarketowson** [comment]
OK thanks @dennisreimann 

I understand from what you have said and how to customize LND settings which are not available as environment variables by creating the docker-compose-generator/docker-fragments/opt-lnd-config.custom.yml file and including the settings you have specified.  That makes sense to me I can see exactly how to do that.

I'm running Core Lightning however on my nodes.  I have the docker installation of BTCPay Server.

So I think what I need to know is - 

1. The file system or docker location of the core lightning config file in the docker version of BTCPay Server.
2. The name of the custom fragment file I need to create under docker-compose-generator/docker-fragments.  Would it be something like: opt-clightning-config.custom.yml ?
3. The exact settings in that config file for core lightning to do the same thing as above that you did with lnd.  

Would the file opt-clightning-config.custom.yml look something like:

version: '3'
services:
  clightning:
    environment:
      CLIGHTNING_EXTRA_ARGS: |
        protocol.wumbo-channels=true
        maxchansize=10000000

I'm just not completely clear on the exact process for doing this with c-lightning given my c-lightning nodes are not wumbo enabled and I need to make them so.

Best Regards,

Clarke

**Reply by dennisreimann** [reply]
That would be a `opt-clightning-config.custom.yml` file like this:

```yaml
version: '3'
services:
  clightning_bitcoin:
    environment:
      LIGHTNINGD_OPT: |
        large-channels=true
```

But keep in mind, the Core Lightning docs state this:

> As of v23.11, this is the default (and thus, the option is ignored). Previously if you didn't specify this, channel sizes were limited to 16777215 satoshi.

**Comment by clarketowson** [comment]
Thanks @dennisreimann 

I tried this on my test server and it didn't work unfortunately.  It caused the lightning docker process to restart itself constantly and never start.  

Please confirm if you see this same behaviour.

Best Regards,

Clarke

**Reply by dennisreimann** [reply]
Can you share the log/error message?

**Comment by clarketowson** [comment]
Sure.  Please let me know the location of the log file.  

The output of the docker ps command was every service started except for btcpayserver/lightning:v23.08 which was listed as 

Restarting in n seconds... 

The counter would make it to 60 then it would loop again and just not restart

Best Regards,

Clarke

**Comment by clarketowson** [comment]
Sorry it's this container that keeps restarting when making the change:

7fc2308c6f97   **saubyk/c-lightning-rest:0.10.6**                           "/sbin/tini -g -- ./…"   2 minutes ago   Restarting (1) 23 seconds ago

Not the btcpayserver/lightning:v23.08 container as I originally stated.

removing the opt-clightning-config.custom.yml file and then setting export BTCPAYGEN_ADDITIONAL_FRAGMENTS="opt-save-storage-s" back to what it was and rerunning . ./btcpay-setup.sh -i gets the container back up again

**Reply by dennisreimann** [reply]
Interesting, what do the logs for that container look like? `docker logs -f generated-clightning_bitcoin_rest-1`

**Comment by clarketowson** [comment]
docker logs -f d7339363bf91 produces the below output:

Error: connect ECONNREFUSED /root/.lightning/lightning-rpc
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
Emitted 'error' event on LightningClient instance at:
    at Socket.<anonymous> (/usr/src/app/lightning-client-js.js:80:23)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '/root/.lightning/lightning-rpc'
}
Waiting /root/.nbxplorer/btc_fully_synched to be created...
The chain is fully synched
Reading config file
warn: --- Starting the cl-rest server ---
warn: --- cl-rest api server is ready and listening on 0.0.0.0:3001 ---
warn: --- cl-rest doc server is ready and listening on 0.0.0.0:4001 ---


If I run this command: docker exec -it d7339363bf91 cat /root/.lightning/config
the config file contains the following:


bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=btcpaydev.intjbilling.com:9735
alias=CHILLYCALENDAR3
announce-addr=w2cjerbflribgjajal5xb6awi26x3y3olghphrbrvthy6el7gv5mbqid.onion:9735
rpc-file=/root/.lightning/lightning-rpc

**Reply by dennisreimann** [reply]
At least with the current/latest versions I canno reproduce this. 

The error `Error: connect ECONNREFUSED /root/.lightning/lightning-rpc` indicates a connection problem: The REST API container cannot connnect to the Core Lightning instance — I doubt this is coming from the setting, can you check the Core Lightning logs as well? `docker logs -f btcpayserver_clightning_bitcoin`

The config file content you posted also does not include the `large-channels=true` setting, so I wonder where that problem might come from.

For me, `cat /var/lib/docker/volumes/generated_clightning_bitcoin_datadir/_data/config` shows this after applying the custom config fragment as stated above:

```yaml
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666

large-channels=true

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
```

**Comment by clarketowson** [comment]
docker logs -f btcpayserver_clightning_bitcoin generates this log file on my test BTCPay Server CHILLYCALENDAR3:

2024-02-22T08:22:15.186Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:16.191Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:17.195Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:18.200Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:19.204Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:20.209Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:21.214Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:22.218Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:23.222Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:24.227Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:25.232Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:26.236Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:27.240Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1
2024-02-22T08:22:28.245Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1

On my test BTCPay server CHILLYCALENDAR3 there is a /var/lib/docker/volumes/ directory but not a generated_clightning_bitcoin_datadir/_data/config under it

but the full directory /var/lib/docker/volumes/generated_clightning_bitcoin_datadir/_data/config is present on both my primary and secondary nodes CHILLYCALENDAR (ulysses) and CHILLYCALENDAR2 (rhetus) which are my production nodes.

[root@ulysses ~]# cat /var/lib/docker/volumes/generated_clightning_bitcoin_datadir/_data/config
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=btcpay.intjbilling.com:9735
alias=CHILLYCALENDAR
announce-addr=swyel3o5czb25iobdvrlig3vzqsgxhkqjb37jgderuouzitdbhhn5cqd.onion:9735
rpc-file=/root/.lightning/lightning-rpc

[root@rhetus ~]# cat /var/lib/docker/volumes/generated_clightning_bitcoin_datadir/_data/config
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=btcpay2.intjbilling.com:9735
alias=CHILLYCALENDAR2
announce-addr=r4mrelvl6zvcjkafcbb7q7vay3xk24ljhyz6t22qvqladog7mzzlueqd.onion:9735
rpc-file=/root/.lightning/lightning-rpc

so yes i'm definitely missing large-channels=true in both config files.

If I add that setting manually to that file on each of my nodes without going through the fragments process will that work?

Best Regards,

Clarke

**Reply by dennisreimann** [reply]
> UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292 0 exited with status 1

This error indicates that the sync process got stuck on that particular block. Does NBXplorer also report a problem with this block? `docker logs -f generated_nbxplorer_1`

If so, you could try to fix it with these commands:

```bash
bitcoin-cli.sh invalidateblock "0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292"
bitcoin-cli.sh reconsiderblock "0000000000000000002033db97de1bf2ed580c79789e8046cf6af03fd0fd3292"
```

The config files you posted lack the `large-channels` setting, so I assume its most likely thge stuck sync problem here.

However, if you still want to add the setting, you can follow [this guide](https://docs.btcpayserver.org/FAQ/LightningNetwork/#how-to-edit-lightningconfig) which I added yesterday. I assume you might have missed the `. ./btcpay-setup.sh -i` after adding the fragment.

> If I add that setting manually to that file on each of my nodes without going through the fragments process will that work?

If you SSH into the container, update it there and restart the container it should work for that instance of the container. I'd rather go with the approach of applying it via the fragments though.

**Comment by clarketowson** [comment]
Thanks for these tips @dennisreimann very much appreciated.  I will take a look at that stuck block issue and I will do that.

I can confirm I did run  . ./btcpay-setup.sh -i  after adding the fragment.

Thanks for posting the guide on the BTCPay website for the clightning fragment that's great.  

I tried again carefully setting the details in my custom fragments file including my alias and rgb colour and I added the large-channels=true setting to the file as well.  Reran btcpay-setup.sh -i but upon restarting the same issue occurs with the stuck restarting the saubyk/c-lightning-rest:0.10.6 container.   Restarting via systemctl restart btcpayserver has no effect either the c-lightning-rest still stays stuck in the restart loop.

Should we bring the issue to saubyk's attention over at Lighting Labs and get his thoughts on it?

With Best Regards,

Clarke

**Comment by clarketowson** [comment]
Hi @dennisreimann - I think I may have figured out the problem.

Under the LIGHTNINGD_OPT: | section I was adding large-channels=true

I tried adding large-channels without =true

This worked.

My lightning config file now looks like this:

[root@redadmiral btcpayserver-docker]# cat /home/docker/volumes/generated_clightning_bitcoin_datadir/_data/config
bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666
large-channels

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=btcpaydev.intjbilling.com:9735
alias=CHILLYCALENDAR3
announce-addr=w2cjerbflribgjajal5xb6awi26x3y3olghphrbrvthy6el7gv5mbqid.onion:9735
rpc-file=/root/.lightning/lightning-rpc

I had the idea after seeing this post: https://github.com/ElementsProject/lightning/issues/7054 which showed a lightning config file with large-channels and not large-channels=true

The saubyk/c-lightning-rest:0.10.6 docker container now stays up and does not go into a constant restarting cycle.

I simply added largechannels to the bitcoin-clightning.yml file and then ran . ./btcpay-setup.sh -i 

 LIGHTNINGD_OPT: |
        bitcoin-datadir=/etc/bitcoin
        bitcoin-rpcconnect=bitcoind
        experimental-offers
        database-upgrade=true
        rpc-file-mode=0666
        large-channels

it would have worked as a custom fragment as well.

The only question now is if large channels really is working on my test node or if the option got ignored.  I might try buying a 20 million sats incoming channel from a liquidity provider to see.

I know large-channels has been deprecated in v23.11 and it's the default and thus the setting will be ignored but I'm running lightning v23.08 on the latest version of BTCPay Server.

Thank you for your help on this very much appreciated.

Kind Regards,

Clarke

**Comment by Kukks** [comment]
By doing this without a custom fragment, the update script will not work.

**Comment by clarketowson** [comment]
OK thanks @Kukks noted.

I have back tracked everything I did and I have used the custom fragment procedure instead and this has worked.  Note I set largechannels and not largechannels = true

[root@redadmiral docker-fragments]# env | grep ADDI
BTCPAYGEN_ADDITIONAL_FRAGMENTS=opt-save-storage-s;opt-clightningd-config.custom
BTCPAY_ADDITIONAL_HOSTS=


[root@redadmiral docker-fragments]# cat opt-clightningd-config.custom.yml
version: '3'
services:
  clightning_bitcoin:
    environment:
      LIGHTNINGD_OPT: |
        alias=CHILLYCALENDAR3
        rgb=0347b8
        large-channels



cat /home/docker/volumes/generated_clightning_bitcoin_datadir/_data/config

bitcoin-datadir=/etc/bitcoin
bitcoin-rpcconnect=bitcoind
experimental-offers
database-upgrade=true
rpc-file-mode=0666

alias=CHILLYCALENDAR3
rgb=0347b8
large-channels

proxy=tor:9050

bind-addr=0.0.0.0:9735
network=bitcoin
announce-addr=btcpaydev.intjbilling.com:9735
alias=CHILLYCALENDAR3
announce-addr=w2cjerbflribgjajal5xb6awi26x3y3olghphrbrvthy6el7gv5mbqid.onion:9735
rpc-file=/root/.lightning/lightning-rpc



docker logs -f 77db8ed16f33

Waiting /root/.nbxplorer/btc_fully_synched to be created...
The chain is fully synched
Reading config file
warn: --- Starting the cl-rest server ---
warn: --- cl-rest api server is ready and listening on 0.0.0.0:3001 ---
warn: --- cl-rest doc server is ready and listening on 0.0.0.0:4001 ---

docker ps indicates the  saubyk/c-lightning-rest:0.10.6 container is up.

Thanks again for the help from the both of you very much appreciated.

Best Regards,

Clarke

**Reply by dennisreimann** [reply]
Ok, glad it's finally working for you. Can we close this now?

**Reply by clarketowson** [reply]
Yes please close this now thank you :-)

**Comment by MytiMouse1** [comment]
Verify
